openshift/machine-config-operator
# MCO-1016: Only update when there is an actual controller change

**- What I did**
Ensure that the subcontrollers for template MC, render MC, secret, and controller config are only updating when there is an actual change. This helps remove unnecessary logs and updates in the controllers.
This is a follow up of https://github.com/openshift/machine-config-operator/pull/536 and has been adapted from https://github.com/openshift/machine-config-operator/pull/4156 to get this conpleted.

**- How to verify it**
Observe the logs in the controllers to ensure there are no update logs when an update hasn't happened.

**- Description for the changelog**
Remove unnecessary update logs for some of the subcontrollers.
